component/client/chat/ChatHistory.tsx

```diff
@@ -57,8 +57,15 @@ class ChatHistory extends React.Component<IHistoryProps, IHistoryState> {
   public componentWillReceiveProps(nextProps: IHistoryProps): void {
     // History receives a new message. Send the message and flag to scroll
     // to the bottom of the component.
-    this.roomService.pushMessage({userFrom: this.props.username, message: nextProps.pushMessage});
-    this.shouldScroll = true;
+    if (nextProps.pushMessage) {
```

Comment:
What does pushMessage do?
This is such a convoluted way of saying messenger gets some message.
The roomService should be included in ClientChat, so that there is no such convoluted way of message passing